### PR TITLE
querier: Use select params for mint and maxt

### PR DESCRIPTION
**Pull request by gouthamve**
TLDR; So turns out, if there was query like `up offset 4w` but `start-end = 15s`, we would end up getting all the chunks from `start-4w` to `end`. This PR makes sure only the relevant chunks are requested.

This is because when we create a `Querier` due to TSDB constraints (and maybe for other reasons), we need a single querier for a query. Which means if we `X / Y` where `X` has `offset 4w` and `Y` doesn't, the `q.minTime`, and `q.maxTime` will need to span the entire `4w`.

But this is mitigated by setting the appropriate parameters on `SelectParams` which this PR makes sure we use.

Relevant promQL code:
https://github.com/prometheus/prometheus/blob/master/promql/engine.go#L465-L467
https://github.com/prometheus/prometheus/blob/master/promql/engine.go#L472
https://github.com/prometheus/prometheus/blob/master/promql/engine.go#L491-L495

### CODE REVIEW

File: pkg/querier/chunk_store_queryable.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ type chunkStoreQuerier struct {
 	mint, maxt        int64
 }
 
-func (q *chunkStoreQuerier) Select(_ *storage.SelectParams, matchers ...*labels.Matcher) (storage.SeriesSet, error) {
-	chunks, err := q.store.Get(q.ctx, model.Time(q.mint), model.Time(q.maxt), matchers...)
+func (q *chunkStoreQuerier) Select(sp *storage.SelectParams, matchers ...*labels.Matcher) (storage.SeriesSet, error) {
+	chunks, err := q.store.Get(q.ctx, model.Time(sp.Start), model.Time(sp.End), matchers...)
```
Comment (bboreham) [inline pkg/querier/chunk_store_queryable.go]: should this check that `sp != nil` ?  Maybe it can't be today, but it seems safer to follow the general pattern.

Comment (tomwilkie) [reply pkg/querier/chunk_store_queryable.go]: This is only call by querier.go (Select), which includes that check.  Calling this with nil should panic IMO.  The only "root" queriers are the ones in querier.go and unified_querier.go, where the nil check need to be.

**Comment by gouthamve** [comment]
This has been running in prod over the last couple of days and everything looks good!
